sigle/src/modules/publicHome/components/PublicHome.tsx

```diff
@@ -100,13 +101,17 @@ const PublicHomeSiteUrl = ({ siteUrl }: { siteUrl: string }) => {
   );
 };
 
-interface PublicHomeProps {
+interface ProfilePageContentProps {
   file: StoryFile;
   settings: SettingsFile;
   userInfo: { username: string; address: string };
 }
 
-export const PublicHome = ({ file, settings, userInfo }: PublicHomeProps) => {
+const ProfilePageContent = ({
```

Comment:
It's really hard to review the changes made to this file, what was the reason for having a new component here? Do you think you could revert back to just one component to make the review simpler and maybe split it into another pull request if needed :)?

Reply:
Ah right yes this was done to wrap the page content in the dashboard layout if the user is visiting their own profile, otherwise not showing the dashboard layout if you are visiting another blog. But yeah sure can do that :)

Reply:
It seems that with the latest change, I am now seeing the menu on another user profile.

![Screenshot 2022-07-22 at 11 35 48](https://user-images.githubusercontent.com/5749437/180411132-dbd787d5-e968-4461-b353-f490406e4cae.png)

This is a logic you were using before.

```
{userInfo.username !== user?.username ? (
        <ProfilePageContent
          file={file}
          settings={settings}
          userInfo={userInfo}
        />
      ) : (
        <DashboardLayout>
          <ProfilePageContent
            file={file}
            settings={settings}
            userInfo={userInfo}
          />
        </DashboardLayout>
      )}
```

Another option to keep the diff clean can be to do smth like :

```
const Layout = userInfo.username !== user?.username ? React.Fragment : DashboardLayout;

<Layout>
// Code
</Layout>
```

Reply:
ohh of course! hadn't thought about this, makes a lot more sense 😅 thanks!